R/survival_cov_adj.R

```diff
@@ -126,6 +128,16 @@ h_strat_derived_outcome_vals <- function(theta, df, treatment, time, status, str
     covariates = covariates,
     n = n
   )
+  strata_number <- seq_along(df_with_outcomes_split)
+  df_with_outcomes_split <- Map(
+    function(df_stratum, stratum_number) {
+      df_stratum$.stratum <- stratum_number
+      df_stratum
+    },
+    df_with_outcomes_split,
+    strata_number
+  )
+  do.call(rbind, df_with_outcomes_split)
```

Comment:
seems unnecessary to split and merge back; just use .stratum = interaction(df[strata]) for the whole dataset

Reply:
Well we have to calculate the derived outcomes separately per stratum. This was like that before this PR already and needs to stay like that. Now we just add the stratum number to each of the produced data sets. So I don't think there is something unnecessary here?

Reply:
oh I see. no problem at all